packages/plugins/sentry/server/middlewares/sentry-performance.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const { stripUrlQueryAndFragment } = require("@sentry/utils");
+
+/**
+ * Programmatic sentry-performance middleware. We do not want to expose it in the plugin
+ * @param {{ strapi: import('@strapi/strapi').Strapi }}
+ */
+module.exports = ({ strapi }) => {
+  const sentryService = strapi.plugin('sentry').service('sentry');
+  sentryService.init();
+  const Sentry = sentryService.getInstance();
+
+  if (!Sentry || !sentryService.hasTracingEnabled()) {
+    // initialization failed or tracing is not enabled
+    return;
+  }
+
+  strapi.server.use(async (ctx, next) => {
+    return new Promise((resolve, reject) => {
```

Comment:
Why are we instantiating a promise here?

Reply:
good point, will just remove these wrapping promises!